quadrature-encoder/src/encoder/incremental.rs

```diff
@@ -38,16 +45,24 @@ where
     T: Zero,
 {
     /// Creates an incremental encoder driver for the given pins.
-    pub fn new(pin_clk: Clk, pin_dt: Dt) -> Self
+    /// NOTE: eh1 requires mutable pin references, but eh0 does not, which upsets clippy sometimes.
+    #[allow(unused_mut)]
```

Comment:
> NOTE: eh1 requires mutable pin references, but eh0 does not, which upsets clippy sometimes.

In that case we should be able to limit the `allow(…)` to just `eh0`, this limiting its scope, I think? 

```suggestion
    #[cfg_attr(not(feature = "eh1"), allow(unused_mut))]
```